modules/kubientBidAdapter.js

```diff
@@ -0,0 +1,147 @@
+  gvlid: VENDOR_ID,
+  supportedMediaTypes: [ BANNER, VIDEO ],
+  isBidRequestValid: function (bid) {
+    return !!(bid && bid.params);
+  },
+  buildRequests: function (validBidRequests, bidderRequest) {
+    if (!validBidRequests || !bidderRequest) {
+      return;
+    }
+    return validBidRequests.map(function (bid) {
+      let floor = 0.0;
+      if (typeof bid.getFloor === 'function') {
+        const mediaType = (Object.keys(bid.mediaTypes).length == 1) ? Object.keys(bid.mediaTypes)[0] : '*';
+        const sizes = bid.sizes || '*';
+        const floorInfo = bid.getFloor({currency: 'USD', mediaType: mediaType, size: sizes});
+        if (typeof floorInfo === 'object' && floorInfo.currency === 'USD' && !isNaN(parseFloat(floorInfo.floor))) {
+          floor = parseFloat(floorInfo.floor);
+        }
+      }
+
+      let adSlot = {
+        bidId: bid.bidId,
+        zoneId: bid.params.zoneid || '',
+        floor: floor || 0.0
+      };
+
+      if (bid.mediaTypes.banner) {
+        adSlot.banner = bid.mediaTypes.banner;
+      }
+
+      if (bid.mediaTypes.video) {
+        adSlot.video = bid.mediaTypes.video;
+      }
+
+      if (bid.schain) {
+        adSlot.schain = bid.schain;
+      }
+
+      let data = {
+        v: VERSION,
+        requestId: bid.bidderRequestId,
+        adSlots: [adSlot],
+        tmax: bidderRequest.timeout,
+        gdpr: (bidderRequest.gdprConsent && bidderRequest.gdprConsent.gdprApplies) ? 1 : 0,
+        consentGiven: kubientGetConsentGiven(bidderRequest.gdprConsent),
+        uspConsent: bidderRequest.uspConsent
+      };
+
+      if (bidderRequest.refererInfo && bidderRequest.refererInfo.referer) {
+        data.referer = bidderRequest.refererInfo.referer
+      }
+
+      if (bidderRequest.gdprConsent && bidderRequest.gdprConsent.consentString) {
+        data.consent = bidderRequest.gdprConsent.consentString
+      }
+
+      return {
+        method: 'POST',
+        url: END_POINT,
+        data: JSON.stringify(data)
+      };
+    });
+  },
+  interpretResponse: function interpretResponse(serverResponse, request) {
+    if (!serverResponse || !serverResponse.body || !serverResponse.body.seatbid) {
+      return [];
+    }
+    let bidResponses = [];
+    serverResponse.body.seatbid.forEach(seatbid => {
+      let bids = seatbid.bid || [];
+      bids.forEach(bid => {
+        const bidResponse = {
+          requestId: bid.bidId,
+          cpm: bid.price,
+          currency: bid.cur,
+          width: bid.w,
+          height: bid.h,
+          creativeId: bid.creativeId,
+          netRevenue: bid.netRevenue,
+          ttl: bid.ttl,
+          ad: bid.adm,
+          meta: {}
+        };
+        if (bid.meta && bid.meta.adomain && utils.isArray(bid.meta.adomain)) {
+          bidResponse.meta.advertiserDomains = bid.meta.adomain;
+        }
+        if (bid.mediaType === VIDEO) {
+          bidResponse.mediaType = VIDEO;
+          bidResponse.vastXml = bid.adm;
+        }
```

Comment:
The test zoneid in your markdown is not returning mediaType with VIDEO in a video request, could this be fixed?

Reply:
@ncolletti I've found the issue. Work in progress.

Reply:
@ncolletti It is fixed.
Test page is [here](https://17slonov.ru/static/video/).

Reply:
Thanks, you are receiving an error from Prebid currently when registering your bidResponse. This is because you're attempting to use instream which needs a vastUrl properly, not xml. Could you make this update?

Reply:
What shall we do if we have VAST XML on our backend? So we need to prepare some solution to save VAST and provide it via URL in vastUrl field.
Also as I understand for instream we have to provide vastUrl and for outstream we have to provide creative in vastXml.
Am I correct ?

Reply:
Apologies this was my confusion, I tested again on the test/instream.html page built with an instream unit auction and it displays properly using the pbjs buildVideoUrl function.